pilot/pkg/bootstrap/server.go

```diff
@@ -1334,3 +1341,27 @@ func (s *Server) serveHTTP() error {
 	s.httpAddr = httpListener.Addr().String()
 	return nil
 }
+
+func serviceUpdateNeedsPush(prev, curr *model.Service) bool {
+	if prev == nil {
+		return true
+	}
+	// If current/previous metadata has "*istio.io" label/annotation, just push
+	for label := range prev.Attributes.Labels {
+		if strings.Contains(label, "istio.io") {
+			return true
+		}
+	}
+	for label := range curr.Attributes.Labels {
+		if strings.Contains(label, "istio.io") {
+			return true
+		}
+	}
+	// Exclude labels and compare so that we do not push on unwanted label changes.
```

Comment:
Are you sure this is safe ? We have plenty of configs based on label selection - and if a label used in any such config is changed, we may need to push.  For example DestinationRule may have label selectors that change the cluster/endpoint composition.

Reply:
I agree with this

Reply:
That is a good point. I will remove this exclude labels logic.